### PR TITLE
feat(searchBox): ability to be non-instant

### LINKED ISSUE #458: Non-instant search box
I think we should also provide our users the ability to not trigger the search at each keystroke but rather once `<Enter>` has been pressed.

I was thinking about patching the `searchBox` widget with something like:

```
search.addWidget(
  instantsearch.widgets.searchBox({
    container: '#q',
    // [...]
    instant: false // will only trigger the search when <Enter> is pressed
  })
);
```

### PR DESCRIPTION
**Pull request by redox**
If `instant` is set to false, the search is only triggered if ENTER is pressed.

 - Added some tests on even registration
 - Removed an outdated version of the searchbox that was at the root level of the widgets/ directory.
 - Updated jsdoc

Fix #458

### CODE REVIEW

File: widgets/search-box/search-box.js [hunk 1]
```diff
@@ -5,6 +5,8 @@ let forEach = require('lodash/collection/forEach');
 let bem = require('../../lib/utils').bemHelper('ais-search-box');
 let cx = require('classnames');
 
+const KEY_ENTER = 13;
```
Comment (vvo) [inline widgets/search-box/search-box.js]: Does it handles both keypad enter and the main enter?

**Comment by vvo** [comment]
I like the feature but I am unsure about `instant` API option. The library is called "instantsearch" but we have an option to disable the "instant": it was my first reaction.

**Comment by vvo** [comment]
Along with this option we would need a `submit` button to be generated (as an option too), because otherwise it makes sense only on desktop

**Comment by redox** [comment]
> I like the feature but I am unsure about instant API option. The library is called "instantsearch" but we have an option to disable the "instant": it was my first reaction.

Any idea what could be a better naming? `searchAsYouType: true/false` was not nice :/

> Along with this option we would need a submit button to be generated (as an option too), because otherwise it makes sense only on desktop

Hmmm, it is the goal responsibility of the searchbox? Because having a submit button means you have a wrapping `<form>` as well. Good point but not sure how to fix it :/

**Comment by vvo** [comment]
> Hmmm, it is the goal responsibility of the searchbox? Because having a submit button means you have a wrapping <form> as well. Good point but not sure how to fix it :/

Well we can have a wrapping form or not, that's not mandatory. You can add just a button type submit and handle the submit on it. Or we can have a wrapping form, we have one for example in the priceRanges widget.

But that's for later

**Comment by vvo** [comment]
> Any idea what could be a better naming? searchAsYouType: true/false was not nice :/

`searchOnEnterKeyPressOnly` exhaustive he!

File: widgets/search-box/search-box.js (hunk 1 above)
Comment (redox) [inline widgets/search-box/search-box.js]: > Does it handles both keypad enter and the main enter?

Sure!

**Comment by redox** [comment]
Updated @vvo

**Comment by pixelastic** [comment]
Sorry I'm late, but as @vvo asked, why do we need that? Shouldn't we push the instant search as the only real way to do intuitive search? I feel like integration this small option (which is in opposition with the main goal of the library) will create an large array of side effects.

**Pull request merged**